R/req-cache.R

```diff
@@ -143,17 +145,22 @@ cache_post_fetch <- function(req, resp, path = NULL) {
 }
 
 cache_body <- function(req, path = NULL) {
-  body <- cache_get(req)$body
+  cache_resp <- cache_get(req)
+  body <- cache_resp$body
+  content_type <- resp_header(cache_resp, "content-type")
```

Comment:
Hmmmm, my mental model was that the response would contain the same headers regardless of whether or not it was cached. Since that's clearly not true, it might be worth doing some more reading about HTTP caching to figure out if we should also be caching and restoring some or all of the headers from the initial response.

Reply:
As usual [stackoverflow](https://stackoverflow.com/a/31107285/7529482) helps 😄 

From [RFC 7232 §4.1][2] :
> The server generating a `304` response **MUST** generate any of the   
> following header fields that would have been sent in a `200 (OK)`   
> response to the same request: `Cache-Control`, `Content-Location`, `Date`,  
> `ETag`, `Expires`, and `Vary`.
> 
> Since the goal of a `304` response is to minimize information transfer
> when the recipient already has one or more cached representations,
> a    sender **SHOULD NOT** generate representation metadata other than the
> above listed fields unless said metadata exists for the purpose of   
> guiding cache updates (e.g., `Last-Modified` might be useful if the   
> response does not have an `ETag` field).

  [1]: https://www.rfc-editor.org/rfc/rfc7232#section-3
  [2]: https://www.rfc-editor.org/rfc/rfc7232#section-4.1